### ISSUE TITLE
API key invalid value

### ISSUE THREAD

**Issue by delaho** [Casvt/Kapowarr#113]
Any idea why this would be happening? Kapowarr could be running beta 3 or alpha and on either one I'll have it working, then I assigned a new volume for Kapowarr in Docker and this happens. Then I could remove the Volume in docker and this still happens. Its weird and doesnt seem dependent on anything on my side.

[2023-12-08 22:30:14][waitress-7][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:14][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:14][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:14][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:14][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:14][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:15][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:15][waitress-7][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:15][waitress-7][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:15][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:15][waitress-5][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:15][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:15][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:15][waitress-1][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:15][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:16][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:16][waitress-6][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:16][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:16][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:16][waitress-7][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:16][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:16][waitress-1][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:16][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:16][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:17][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:17][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:17][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:17][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:17][waitress-1][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:17][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:17][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:32][MainThread][INFO] Starting up Kapowarr
[2023-12-08 22:30:32][MainThread][INFO] Kapowarr running on http://0.0.0.0:5656/
[2023-12-08 22:30:35][waitress-5][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:35][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:36][waitress-7][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:36][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:36][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:36][waitress-6][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:36][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:36][waitress-1][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:36][waitress-5][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:36][waitress-7][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:36][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:37][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:37][waitress-6][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:37][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:37][waitress-5][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:37][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:37][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:37][waitress-6][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:37][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:37][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:38][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:38][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:38][waitress-5][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:38][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:38][waitress-6][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:38][waitress-7][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:38][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:39][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:39][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:39][waitress-7][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:39][waitress-1][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:39][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:39][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:39][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:39][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:40][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:40][waitress-6][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:40][waitress-5][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:40][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:40][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:40][waitress-7][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:40][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:40][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:40][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:41][waitress-6][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:41][waitress-1][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:41][waitress-5][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:41][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:41][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:41][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:41][waitress-5][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:41][waitress-6][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:41][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:42][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:42][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:42][waitress-9][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:42][waitress-0][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:42][waitress-5][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:42][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:42][waitress-4][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:42][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:42][waitress-8][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:43][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:43][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:43][waitress-2][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:43][waitress-3][WARNING] This key in the API request has an invalid value: api_key = 1fc322fb1cd9e8766d24d69ec38ca7fc
[2023-12-08 22:30:43][MainThread][INFO] Shutting down Kapowarr...

**Comment by Casvt** [comment]
I'm assuming that you're switching between multiple Kapowarr instances? I think there is some problem with that because for each instance, there is a separate API key. When you access the other one, it tries to use the same API key, but it's invalid because it's a whole different instance.

So there should be a fix that tries to authenticate when an API key is set, but turns out to be invalid.

**Comment by delaho** [comment]
The thing is, I always stopped my other instance and yet this persisted even though I only ran 1 at a time.

**Comment by Casvt** [comment]
Yeah but that doesn't matter. I know what the problem is, it's an easy fix.

**Comment by Cantello** [comment]
I have the same issue here even with one instance and on a fresh docker install. I store the db in a mapped volume and deleted the volume prior to changing paths in the compose file and restarting the container. Issue still persists.
The webUI is constantly redirecting from IP:5656/login?redirect=/ and similar error messages as above are in the docker container's log.
Any way to get a nightly version as it already seems to be fixed?

**Comment by Casvt** [comment]
Yeah two instances at the same time or two different instances after each other is the same. You can pull from `mrcas/kapowarr-alpha:latest` for alpha releases. The last few ones have the fix included.

**Comment by Cantello** [comment]
Thanks, that worked!

**Comment by ChrisBen191** [comment]
This issue occurred when with one instance of Kapowarr. It occurred when I shut down the docker container, updated compose file  to add another root folder (manga), and then restarted the service. I too saw the webUI constantly redirecting from IP:5656/login?redirect=/ and similar error messages for the API key. Can this be resolved even when I have one instance of Kapowarr?

**Comment by ChrisBen191** [comment]
The "docker" user created also has access to all relevant folders (manga and comics). Is having multiple root folders what you describe as multiple instances?

**Comment by Casvt** [comment]
@ChrisBen191 multiple root folders is not multiple instances. It's running two Kapowarrs side by side or changing the database and then running on the same host and IP.

The fix is available in the alpha release (`mrcas/kapowarr-alpha:latest`).

P.S. Kapowarr does not have any special support for manga, so it most likely won't work for it.

**Comment by ChrisBen191** [comment]
@Casvt thanks for the clarification, I switch as recommended and I'm now successfully operating Kapowarr. Thanks!
